modules/geolocation/geolocation.py

```diff
@@ -129,3 +129,38 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+
+    def map_location_from_pixel(self, pixels, coordinates):
```

Comment:
Please create a separate branch and pull request for the geolocation task.

In general each task should be separated into its own branch, and when ready for review, its own pull request.